pkg/aspect/configError.go

```diff
@@ -77,5 +77,19 @@ func (e ConfigError) Error() string {
 }
 
 func (e *ConfigErrors) Error() string {
+	if e == nil || e.Multi == nil {
+		return ""
+	}
 	return e.Multi.Error()
 }
+
+// Extend joins 2 configErrors together.
+func (e *ConfigErrors) Extend(ee *ConfigErrors) *ConfigErrors {
```

Comment:
The reason for ConfigErrors is that we wanted a distinct method signature to help adapter authors know what kind of error to return. Additionally, the methods on ConfigErrors are explicitly designed to encourage the right pattern.